src/main/java/org/sonarlint/intellij/promotion/PromotionProvider.kt

```diff
@@ -172,12 +182,14 @@ class PromotionProvider(private val project: Project) {
         }
     }
 
-    private fun processExtraLanguagePromotion(notifications: SonarLintProjectNotifications, extension: String) {
-        val language = findLanguage(extension, extraEnabledLanguagesInConnectedMode)
+    private fun processExtraLanguagePromotion(notifications: SonarLintProjectNotifications, languagesToPromote: Set<Language>) {
+        val languages = findExtraLanguages(languagesToPromote, extraEnabledLanguagesInConnectedMode)
 
-        if (language != null) {
-            showPromotion(notifications, "Enable " + language.label + " analysis by connecting your project")
-        }
+        if (languages.isEmpty()) return
+
+        val languagesText = languages.joinToString(", ")
```

Comment:
Maybe to stay consistent with SLE, we can do the following:
```suggestion
        val languagesText = languages.joinToString(" / ")
```